### ISSUE TITLE
Hide HVAC mode selection if only one HVAC mode available

### ISSUE THREAD

**Issue by giggo** [nielsfaber/scheduler-card#127]
My/some thermostats have only one hvac_modes:
<img width="600" alt="grafik" src="https://user-images.githubusercontent.com/213763/95847461-05730900-0d4d-11eb-94b2-11460d29e725.png">

Maybe it would make sense for the scheduler card to not even offer the mode selection option in case there is only one mode.

<img width="457" alt="grafik" src="https://user-images.githubusercontent.com/213763/95847865-8c27e600-0d4d-11eb-8409-bb846d26c7ed.png">

**Comment by nielsfaber** [comment]
Good point!
I will implement this.
Strictly speaking the problem is with your thermostat integration.
If there are less than 2 modes available, the thermostat does not support setting the mode (a setting with one fixed option is not really a setting)..

**Comment by nielsfaber** [comment]
By updating to the latest (v1.8.1) release of the card it should be fixed.

**Comment by nielsfaber** [comment]
@giggo Small question:
Did you ever try to schedule a set_temperature action with this entity.
I see that it only supports hvac_mode auto.
Which suggests that the temperature setpoint is controlled by the device itself, and set_temperature action is ignored.

I am asking because some people have issues where the scheduler does not trigger any (heating) action.
And they also are missing the hvac_mode heat.
I'm trying to make some sense of it..

**Comment by andreipopovici** [comment]
@nielsfaber There seems to be an issue with the Generic Thermostat implementation outside of scheduler's scope, if that's what you mean.

However, from a usability perspective, it might be better to have two separate actions for scheduling a temperature change and scheduling a `hvac_mode` change, instead of using `set_temperature` and trying to do both in one step. I think most novice users understand "I want my thermostat to turn on at this hour" rather than "I want to set temperature and also start my thermostat at this hour".

**Comment by nielsfaber** [comment]
@andreipopovici 

> I think most novice users understand "I want my thermostat to turn on at this hour"

In HA there is no such thing as turning on a thermostat. You need to set either a temperature or enable a preset or set it to auto mode.

In the beginning, `set_temperature` and `set_hvac_mode` were different actions in the card.
I got complaints from people who said that `set_temperature` didn't have any effect on their thermostats.
After investigation, it was found that their thermostats needed to be set to `hvac_mode=heat` together with the temperature setpoint.
If their thermostats were still in `off` or `auto`, the `set_temperature` command would be completely ignored.

So now, if your thermostat supports `hvac_mode=heat` the card will give you a 'heat' action (instead of 'set_temperature').
The heat action makes sure that `hvac_mode` is set to `heat` together with the temperature setpoint.
This is exactly how the service call is intended to be [configured in HA](https://www.home-assistant.io/integrations/climate/#service-climateset_temperature).

> However, from a usability perspective, it might be better to have two separate actions for scheduling a temperature change and scheduling a hvac_mode change, instead of using set_temperature and trying to do both in one step.

The heat action is actually an improvement in terms of usability. Because you don't have to care anymore about having your thermostat set to the correct `hvac_mode`.
And since the card doesn't allow you to combine multiple actions for the same timeslot, this saved users from having to make a duplicate (synchronised) schedule that handles the second action.

Bottom line: if your thermostat does have `hvac_mode=heat` but the 'heat' action of the card does not work, this is a problem in your climate integration.
I cannot unite both worlds...

**Comment by andreipopovici** [comment]
@nielsfaber You have a point. I agree that you made the correct design decision.

> This is exactly how the service call is intended to be configured in HA.

> Bottom line: if your thermostat does have hvac_mode=heat but the 'heat' action of the card does not work, this is a problem in your climate integration.

Also true. Unfortunately, as pointed in https://github.com/nielsfaber/scheduler-component/issues/42, the climate integration with this problem is the one provided by HA via Generic Thermostat. Alas, I have no control over that 😢 (as in, I wouldn't know which part of the HA component to open a PR for).

Like @nelim in the other issue, I was able to fix that by adding a custom action to the generic `climate.entity` that only calls `set_hvac_mode`, with the compromise of setting the temperature manually.

(Side note, `scheduler/edit` doesn't yet support updating climate actions, it seems, but it's not worth opening an issue since the docs do mention it's a WIP. I'll just have to recreate the existing schedules.)

I suppose a generic (pun intended!) way to fix this would be the ability to add multiple actions to a single `customize` entry. However, that probably opens up a whole other can of worms, which HA automations already opened.

Maybe a docs update showing an example of setting up a script or automation and adding it as a custom action via `customize` would help newbies the most.

**Comment by nielsfaber** [comment]
>  with the compromise of setting the temperature manually.

You can also define a conventional 'set temperature' action right?
```
customize:
  climate:
    actions:
      - service: set_temperature
         variable:
           field: temperature
           min: 10
           max: 30
```` 
Add icons and names as you want..

(or is this what you mean by 'manually')

> (Side note, scheduler/edit doesn't yet support updating climate actions, it seems, but it's not worth opening an issue since the docs do mention it's a WIP. I'll just have to recreate the existing schedules.)

scheduler/edit is used by the card as well, for any change you make. It supports all features you see in the card, but the syntax is not really 'public'. 
Fun fact: you can even create a super fancy schedule that targets multiple entities and/or multiple actions per timeslot.
I do not recommend this though, because the card is not able to display these cases.

> the ability to add multiple actions 

I know.. I know.. 
But HA users are greedy! Keeping me busy all the time! And panic if changes introduce bugs 😅 
If I find a nice way to introduce this in the GUI I will do so 👍

**Comment by andreipopovici** [comment]
@nielsfaber I agree we're greedy little feature requesters 🤣 Luckily, your component is gaining popularity and will probably be included in HA core soon, which will take some pressure off.

> (or is this what you mean by 'manually')

In my case, it's just easier to set a fixed temperature via Lovelace and just toggle `hvac_mode` to control the generic thermostat (it's hooked up to a dumb electric space heater via an IR remote). Your suggestion implies setting a second, mirrored schedule for the temp, but since it doesn't need to change often, it's redundant (at least until the PR for the generic thermostat fix is approved 😅).

> scheduler/edit is used by the card as well, for any change you make. It supports all features you see in the card, but the syntax is not really 'public'.

After adding the `set_hvac_mode` service in a `customize` section, as per your suggestion in the other thread, when editing an existing schedule and changing from the default `Heat` action to my custom action, I get this error when saving:

![Screen Shot 2020-11-01 at 19 22 33](https://user-images.githubusercontent.com/1143417/97809846-3d98a800-1c78-11eb-86d5-bfd1aaecd846.png)

**Comment by nielsfaber** [comment]
Time to clean up old issues.. Please re-open if the issue is not resolved.
